pkg/virtctl/root.go

```diff
@@ -55,10 +59,98 @@ func NewVirtctlCommand() *cobra.Command {
 	return rootCmd
 }
 
-func Execute() {
+func Execute(args []string) error {
 	log.InitializeLogging("virtctl")
-	if err := NewVirtctlCommand().Execute(); err != nil {
-		fmt.Println(strings.TrimSpace(err.Error()))
-		os.Exit(1)
+	// check whether there is a MIME config file
+	// if yes, then set up the args for commands and execute
+	// otherwise normal operation
+	confArgs, _ := parseMime(args)
+
+	cmd := NewVirtctlCommand()
+
+	if confArgs != nil {
+		// if valid config file was used as single command argument
+		// override the defaults with the commands from config
+		cmd.SetArgs(confArgs)
+	} else {
+		cmd.SetArgs(args[1:])
+	}
+
+	if err := cmd.Execute(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func parseMime(args []string) ([]string, error) {
+	var parsedArgs []string
+
+	filePath, err := checkForFile(args)
+	if err == nil {
+		data, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			return nil, fmt.Errorf("Cannot read config file: %s", filePath)
+		}
+
+		parsedArgs, err = parseMimeConfig(string(data))
+		if err != nil {
+			return nil, err
+		}
 	}
+
+	return parsedArgs, err
+}
+
+func checkForFile(args []string) (string, error) {
+	var err error
+	filePath := ""
+
+	if len(args) == 2 {
+		if strings.HasSuffix(args[1], ".vvv") {
+			exist, _ := util.FileExists(args[1])
+			if exist {
+				filePath = args[1]
+			} else {
+				err = errors.New("File does not exist")
+			}
+
+		} else {
```

Comment:
Think for insance about `virtctl version`. It would not work now.

Reply:
Fixed